libcst/codegen/gen_matcher_classes.py

```diff
@@ -396,8 +399,8 @@ def _get_clean_type_and_aliases(
 
     # First, get the type as a parseable expression.
     typestr = repr(typeobj)
-    if typestr.startswith("<class '") and typestr.endswith("'>"):
-        typestr = typestr[8:-2]
+    typestr = re.sub(CLASS_RE, r"\1", typestr)
+    typestr = re.sub(OPTIONAL_RE, r"typing.Optional[\1]", typestr)
```

Comment:
This won't work for `Union[A, B, NoneType]` or `Union[A, None]`. While this might not cause an issue today, I don't want to be on the debugging end if it pops up in the future :)

Reply:
Why don't it work for `Union[A, B, NoneType]`, my regex rejects that.

I'm not sure I understand the difference between `Union[A, NoneType]` and `Union[A, None]`

Reply:
Hm, fair enough. apparently `Option[Union[A, B]]` always gets flattened to `Union[A, B, NoneType]` by dataclasses, so it doesn't even matter.